File: sdk/elixir/test/direnv_config/version_test.exs
defmodule DirenvConfig.VersionTest do
  use ExUnit.Case, async: true

  alias DirenvConfig.Version

  setup do
    tmp = Path.join(System.tmp_dir!(), "dc_version_test_#{System.unique_integer([:positive])}")
    File.mkdir_p!(tmp)

    on_exit(fn ->
      File.rm_rf!(tmp)
    end)

    {:ok, store: tmp}
  end

  describe "read/1" do
    test "returns 0 when no version file exists", %{store: store} do
      assert Version.read(store) == 0
    end

    test "reads existing version", %{store: store} do
      File.write!(Path.join(store, ".version"), "5")
      assert Version.read(store) == 5
    end

    test "returns 0 for invalid content", %{store: store} do
      File.write!(Path.join(store, ".version"), "garbage")
      assert Version.read(store) == 0
    end
  end

  describe "bump/1" do
    test "bump from zero returns 1", %{store: store} do
      assert {:ok, 1} = Version.bump(store)
      assert Version.read(store) == 1
    end

    test "bump increments existing", %{store: store} do
      File.write!(Path.join(store, ".version"), "7")
      assert {:ok, 8} = Version.bump(store)
      assert Version.read(store) == 8
    end

    test "sequential bumps", %{store: store} do
      assert {:ok, 1} = Version.bump(store)
      assert {:ok, 2} = Version.bump(store)
      assert {:ok, 3} = Version.bump(store)
      assert Version.read(store) == 3
    end
  end
end

File: utilities/shell/direnv-config/sdk/typescript/test/version.test.ts
import { describe, it, expect, beforeEach } from 'vitest';
import { mkdtemp } from 'node:fs/promises';
import { tmpdir } from 'node:os';
import { join } from 'node:path';
import { bumpVersion, readVersion } from '../src/version.js';

describe('bumpVersion', () => {
  let store: string;

  beforeEach(async () => {
    store = await mkdtemp(join(tmpdir(), 'dc-version-'));
  });

  it('bump from zero returns 1', async () => {
    const version = await bumpVersion(store);
    expect(version).toBe(1);
  });

  it('bump increments existing version', async () => {
    await bumpVersion(store); // -> 1
    const version = await bumpVersion(store);
    expect(version).toBe(2);
  });

  it('sequential bumps increment correctly', async () => {
    const v1 = await bumpVersion(store);
    const v2 = await bumpVersion(store);
    const v3 = await bumpVersion(store);
    expect(v1).toBe(1);
    expect(v2).toBe(2);
    expect(v3).toBe(3);
  });

  it('readVersion returns 0 when no .version file exists', async () => {
    const version = await readVersion(store);
    expect(version).toBe(0);
  });
});
